src/cmd/interactive/handler.rs

```diff
@@ -0,0 +1,156 @@
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use std::path::PathBuf;
+use anyhow::Error;
+use futures::TryStreamExt;
+use http::{Method, Request, Response, StatusCode};
+use hyper::server::conn::{Http};
+use hyper::service::Service;
+use hyper::Body;
+use tokio::sync::Mutex;
+use std::thread::JoinHandle;
+use tokio::runtime::Runtime;
+use tracing::instrument;
+use tokio::net::{UnixListener};
+#[cfg(unix)]
+use crate::todarpc::TodaRpc;
+use crate::injector::{InjectorConfig};
+
+#[derive(Debug)]
+pub struct TodaServer {
+    toda_rpc: Arc<Mutex<TodaRpc>>,
```

Comment:
we don't need the mutable reference of `TodaRpc`, remove `Mutex` here.

```suggestion
    toda_rpc: Arc<TodaRpc>,
```